modules/charon-core/src/main/java/org/wso2/charon3/core/encoder/JSONDecoder.java

```diff
@@ -306,6 +303,15 @@ public <T extends AbstractSCIMObject> T decodeResource(String scimResourceString
                     //user may define the attribute by its fully qualified uri
                     attributeValObj = decodedJsonObj.opt(attributeSchema.getURI());
                 }
+                if (attributeValObj == null) {
+                    String attributeSchemaName = attributeSchema.getName().toLowerCase();
+                    for (Iterator it = decodedJsonObj.keys(); it.hasNext(); ) {
+                        String key = (String) it.next();
+                        if (key.toLowerCase().equals(attributeSchemaName)) {
```

Comment:
we could user key.equalsIgnoreCase()

Reply:
Thanks for the hint. Changed it accordingly: 58677aabc9e4aafefbb7dbfe20952f1457a3c1c5